include/xkbcommon/xkbcommon.h.jinja

```diff
@@ -442,15 +451,16 @@ struct xkb_rule_names {
  * @param[in]  size   Size of the buffer.
  *
  * @warning If the buffer passed is too small, the string is truncated
- * (though still NUL-terminated); a size of at least 64 bytes is recommended.
+ * (though still NUL-terminated); a size of at least
+ * `XKB_KEYSYM_NAME_MAX_SIZE` bytes is recommended.
```

Comment:
`XKB_KEYSYM_MAX_SIZE` is computed from the header file, so it will always be the minimal correct value to use. The only issue I see that could happen is it if an app is compiled with an xkbcommon version different from the one it is dynamically linked, _and_ this version has a greater value for `XKB_KEYSYM_NAME_MAX_SIZE`.

But I still think it is better to avoid unnamed numeric constants.

Possible solutions:
- Add a margin: e.g. `XKB_KEYSYM_NAME_MAX_SIZE = <current exact value> + 15`
- Set to current advised length `XKB_KEYSYM_NAME_MAX_SIZE = 64`.

Reply:
> The only issue I see that could happen is it if an app is compiled with an xkbcommon version different from the one it is dynamically linked, and this version has a greater value 

yes, that's the one I'm concerned about because that's pretty much the case for any application that uses a shared library but doesn't get updated regularly.

> But I still think it is better to avoid unnamed numeric constants.

So the thing here is that it's not a numeric constant, it's a buffer size that the application gets to pick.

Reply:
oh, and ftr: if you want this as a *private* `#define` in some utility header for use by our tests/tools that's perfectly fine, I just really dislike this as part of the header API.

Oh, and for explanation: this is a bug that won't hit us for years, if ever. But when it hits we then have several years of applications that are all broken in the same way and hard to fix.

Reply:
I see, thank you to take the time to explain. So I refactored this to be used internally only.